src/components/Image/ImageLoader.jsx

```diff
@@ -20,6 +20,14 @@ class ImageLoader extends React.Component {
     this.loadNextSrc()
   }
 
+  componentWillUnmount() {
+    if (this.img) {
+      this.img.onload = this.img.onerror = null
+      this.img.src = ''
+      clearTimeout(this.timeout)
+    }
+  }
+
```

Comment:
Cleaning up when the component unmounts, most notably canceling the image loading requests that are still pending.